ksql-engine/src/main/java/io/confluent/ksql/KsqlEngine.java

```diff
@@ -312,6 +312,66 @@ public QueryMetadata getQueryExecutionPlan(final Query query, final KsqlConfig k
     }
   }
 
+  public List<Pair<String, Statement>> parseStatements(
+      final String queriesString,
+      final MetaStore tempMetaStore
+  ) {
+    final MetaStore tempMetaStoreForParser = tempMetaStore.clone();
+    final KsqlParser ksqlParser = new KsqlParser();
+    final List<SqlBaseParser.SingleStatementContext> parsedStatements;
+    try {
+      parsedStatements = ksqlParser.getStatements(queriesString);
+    } catch (Exception e) {
+      throw new ParseFailedException("Exception while parsing statements :" + e.getMessage(), e);
+    }
+
+    final List<Pair<String, Statement>> queryList = new ArrayList<>();
+
+    for (SqlBaseParser.SingleStatementContext singleStatementContext : parsedStatements) {
+      try {
+        final Pair<Statement, DataSourceExtractor> statementInfo = ksqlParser.prepareStatement(
+            singleStatementContext,
+            tempMetaStoreForParser
+        );
+        final String statementString = getStatementString(singleStatementContext);
+        Statement statement = statementInfo.getLeft();
+        handleStatement(statementInfo,
+            singleStatementContext,
+            tempMetaStore,
+            tempMetaStoreForParser);
+        queryList.add(new Pair<>(statementString, statement));
+      } catch (Exception e) {
+        throw new ParseFailedException(
+            String.format("Exception while processing statement: %s. Error: %s",
+                getStatementString(singleStatementContext),
+                e.getMessage()),
+            e);
+      }
+
+    }
+    return queryList;
+  }
+
+  private void handleStatement(
+      final Pair<Statement, DataSourceExtractor> statementInfo,
+      final SqlBaseParser.SingleStatementContext singleStatementContext,
+      final MetaStore tempMetaStore,
+      final MetaStore tempMetaStoreForParser) {
+    Statement statement = statementInfo.getLeft();
```

Comment:
nit: final.

(Have you not configured your IDE to warn on this?)

Reply:
Statement can be updated here.